airflow/providers/common/sql/hooks/sql.py

```diff
@@ -197,8 +197,7 @@ def get_pandas_df(self, sql, parameters=None, **kwargs):
                 "'apache-airflow-providers-common-sql[pandas]'."
             )
 
-        with closing(self.get_conn()) as conn:
-            return psql.read_sql(sql, con=conn, params=parameters, **kwargs)
+        return psql.read_sql(sql, con=self.get_sqlalchemy_engine(), params=parameters, **kwargs)
```

Comment:
Better explicit open connection, and close it. I don't think pandas actually do this stuff.
As well as dispose SQLAlchemy Engine

```suggestion
        engine = self.get_sqlalchemy_engine()
        try:
            with engine.connect() as conn:
                return psql.read_sql(sql, con=conn, params=parameters, **kwargs)
        finally:
            engine.dispose()
```

Reply:
Thank you for catching this @Taragolis, I should've looked deeper into the engine to see how the connection works, I made an assumption that it would be handled by sqlalchemy.